WakeUpAlarm/Models/Alarm.swift

```diff
@@ -45,8 +67,18 @@ class Alarm: ObservableObject {
     }
     
     // Takes in a time and updates the upcoming AlarmSettings.
-    func updateUpcomingTime() {
-        
+    func updateUpcomingTime(newDateTime: Date) {
+        data.finalWakeUpTime = newDateTime
+    }
+    
+    func getStartDateTime() -> Date {
+        let totalAlarmDuration = Double(data.timeIntervalLength) * Double(data.alarmFrequency)
+        let alarmStartTime = Calendar.current.date(byAdding: .minute, value: -Int(totalAlarmDuration), to: data.finalWakeUpTime)!
+        return alarmStartTime
+    }
+    
+    func getTimeIntervalLength() -> Double {
+        return data.timeIntervalLength
     }
     
     // Returns whether the alarm
```

Comment:
discussion for future: having a state flag for the alarm being "on"/"off" as I mentioned in the comment for line 60 would probably streamline these couple functions below too